### PR TITLE
A4A: Sites Dashboard - Update Dataviews version to latest

### PR DESCRIPTION
**Pull request by gogdzl**
Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/311

## Proposed Changes

* This PR updates wordpress/dataviews to the latest version (v1.0.0).
  * List of issues fixed:
    * Fixed status filter behavior, which changed in this version
    * Fixed dataviews styling when preview pane is open
    * Fixed scrolling 

## Testing Instructions

* One of our components changed, so run `yarn install` to make sure everything is set-up properly.
* Run `yarn start-a8c-for-agencies`
* Go to http://agencies.localhost:3000/sites
* Verify that the new version of our dashboard's dataview component is functioning properly in A4A. There have been some changes compared to our current version, which are outlined below, along with one known issue that we must disregard during testing.

Changes:

**Filter selection**

Previous Version:
image

Latest:

image

image

**When Preview Pane is open, filters now show multiple lines**

Previews Version:
image

Latest:
image

**Known Issue - Please ignore this one for now**

When the Preview Pane is open and pagination is visible, the page's alignment looks off.

image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by coder-karen — COMMENTED** [review]
Wow, nice work getting that filter dropdown to look as good as it does!

I found a couple of things when reviewing so will drop these here:

It looks like the border around the 'Add filter' button has a missing top border (when highlighted). Same with the Reset link after that has been pressed. It only shows briefly when testing locally - the effect is more visible testing on Horizon from what I can see.

Screenshot 2024-04-23 at 10 06 39


Also, if I click on 'Needs Attention' from the sidebar, it opens up the filter dropdown (it does correctly select the right filter). That seems a bit of an odd experience, probably makes sense to hide that if we can? Maybe for a follow-up?

Another question I have is related to the dataviews related styles. Do we want to keep them all in one place, rather than across files? I would think so, purely because we're trying to decouple Jetpack / A4A styles from Dataviews generally. Most of these style additions look like they'd need to be linked with general Dataviews styles. This PR is relevant anyway - https://github.com/Automattic/wp-calypso/pull/89757/files - @cleacos do you know if it's better to merge one PR before the other and then sort out styles later? For example the decoupling PR first, then styles here can be moved into the new relevant files?

**Comment by cleacos** [comment]
I noticed the same issues that Karen's commented. The top border issue also happens in the pagination option:

image


> This PR is relevant anyway - https://github.com/Automattic/wp-calypso/pull/89757/files - @cleacos do you know if it's better to merge one PR before the other and then sort out styles later? For example, when decoupling PR first, can styles here be moved into the new relevant files?

I'd say this PR needs some refinements or fixes before merging. We already have customers testing A4A, so we cannot proceed with these issues. That said, @coder-karen, your PR will go first, which means that after merging it, we will update any affected style in this PR.

**Review by cleacos — DISMISSED** [review]
Karen comments:

- We have to fix those UX issues and the style ones.

I added some designers as reviewers in case we need some feedback about it or alternatives.

**Comment by cleacos** [comment]
@coder-karen, we have to update this comment too: https://github.com/Automattic/wp-calypso/pull/87979/files updating the lock version `1.0` and telling that we have to keep the `@wordpress/dataviews` package in the `ignoreDeps`. It will be there until the `DataViews` become more stable. We will update the package manually.

File: client/a8c-for-agencies/sections/sites/features/jetpack/jetpack-sites-dataviews.tsx
Comment (cleacos) [inline client/a8c-for-agencies/sections/sites/features/jetpack/jetpack-sites-dataviews.tsx:96]: Also, we have to update this on Dotcom, once the base PR is merged

File: client/a8c-for-agencies/sections/sites/features/jetpack/jetpack-sites-dataviews.tsx
Comment (cleacos) [inline client/a8c-for-agencies/sections/sites/features/jetpack/jetpack-sites-dataviews.tsx:106]: Also, we have to update this on Dotcom, once the base PR is merged

**Comment by coder-karen** [comment]
This should be ready for review again now. The remaining issue (dropdown display when clicking on 'Needs Attention' from the sidebar) has been added to an issue here: https://github.com/Automattic/automattic-for-agencies-dev/issues/351

**Comment by cleacos** [comment]
I'm reviewing this again

**Comment by cleacos** [comment]
I was waiting for the Dotcom team to finish their work before their CfT. On Monday, I'll check everything with the recent updates and I'll ping them for a review too.

**Comment by cleacos** [comment]
**Issues at A4A and Dotcom**

The columns are not hidden
image


**Issues detected at Dotcom:**

image

**Comment by cleacos** [comment]
**Dotcom:**

image

**Comment by allilevine** [comment]
For **Dotcom**, the collapsed sites list is missing:

Screen Shot 2024-04-29 at 4 18 52 PM

**Comment by davemart-in** [comment]
@cleacos are the instructions for testing this on Dotcom similar to testing it on A4A? Sorry, help to test this, but that part wasn't clear to me.

**Comment by jeffgolenski** [comment]
Per the convo in slack: p1714407123786029-slack-C06JY8QL0TU

Recommendation is to hold off merging until after launch on Thursday / until the core component advances a little more. There are several broken pieces atm.

**Comment by davemart-in** [comment]
Thanks! I moved this to our deferred column on the Dotcom side of things.

**Comment by Addison-Stavlo** [comment]
> Thanks! I moved this to our deferred column on the Dotcom side of things.

Note this version bump may enable other items we desired to have for initial release on dotcom @davemart-in , [such as the ability to have selection callbacks across entire rows](https://github.com/WordPress/gutenberg/pull/59803).

**Comment by cleacos** [comment]
- Reabased trunk and conflicts resolved.
- Updated the `DotcomSitesDataviews`

**Comment by cleacos** [comment]
Updated DataViews package to version 1.1.0

**Comment by davemart-in** [comment]
On the Dotcom side, I punted this to after i2. If it lands before then, great! Just trying to clean our board up to get ready for final launch reviews beginning of next week.

**Comment by roo2** [comment]
I was hoping to get another commit in https://github.com/WordPress/gutenberg/pull/61307 but had some delays getting it merged, I think I'll just have to update the package version again once this is merged.

**Comment by cleacos** [comment]
> On the Dotcom side, I punted this to after i2. If it lands before then, great!

For now, this is on hold until we decide what to do. I'm finishing the P2-post to talk about it.

For Dotcom, the DataViews v1.1 breaks it a bit, more than A4A.

**Comment by cleacos** [comment]
Discussion about the DataViews future on: pe7F0s-1OO-p2

**Comment by cleacos** [comment]
New reference: https://github.com/Automattic/jetpack-manage/issues/337

**Pull request closed without merging**
